app/packs/src/decidim/vocdoni/voter/new-vote.js

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable no-console */
+import { VocdoniSDKClient } from "@vocdoni/sdk";
+
+import VoteQuestionsComponent from "./vote_questions.component";
+import VoteComponent from "./setup-vote";
+import PreviewVoteComponent from "./setup-preview";
+
+/*
+ * Mount the VoteComponent object and bind the events to the UI
+ *
+ * @param {object} voteComponent The VoteComponent object
+ *            Can be used for a Preview or a Real Vote
+ * @param {object} $voteWrapper The jQuery Element with the wrapper for the vote
+ * @param {object} questionsComponent The VoteQuestionsComponent object
+ */
+const mountVoteComponent = async (voteComponent, $voteWrapper, questionsComponent) => {
+  await voteComponent.bindEvents({
+    onBindSubmitButton(onEventTriggered) {
+      $(".button.confirm").on("click", onEventTriggered);
+    },
+    onStart() {},
+    onBallotSubmission(validVoteFn) {
+
+      /*
+       * @param {object} formData The jQuery object with the selected answers
+       *
+       * @return {array} An array with the selected answers
+       */
+      const getFormData = (formData) => {
+        /* eslint-disable no-invalid-this */
+        return formData.map(function() {
+          const answerValue = this.value;
+          return Number(answerValue.split("-")[1]);
+        }).get();
+        /* eslint-enable no-invalid-this */
+      };
+      const formData = getFormData($voteWrapper.find(".answer_input:checked"));
```

Comment:
```suggestion
      const $checks = $voteWrapper.find(".answer_input:checked")
      const formData = $checks.map((_idx, el) => el.value.split("-")[1]}).get());
```
I find the use of "this" confuse in this context (as is comming from jquery but not very clear), maybe we could rewrite it to something like this

Reply:
Much clean indeed 😍 

I need to fix a couple details: 

1. There's a leftover closing curly brace there ("}")
>       const formData = $checks.map((_idx, el) => el.value.split("-")[1]**}**).get());

2. we need to convert to Number as that's what the SDK expects. 

I made the changes on your suggestion so I can just accept it, but great refactor!